lib/datasource/index.spec.ts

```diff
@@ -280,4 +280,18 @@ describe(getName(), () => {
     });
     expect(res.sourceUrl).toEqual('https://github.com/Jasig/cas');
   });
+
+  it('applies replacements', async () => {
+    npmDatasource.getReleases.mockResolvedValue({
+      releases: [{ version: '1.0.0' }],
+    });
+    const res = await datasource.getPkgReleases({
+      datasource: datasourceNpm.id,
+      depName: 'abc',
+      replacementName: 'def',
+      replacementVersion: '2.0.0',
```

Comment:
I'm a bit confused as to why we need the datasource involved

Reply:
I thought it would be required to have `packageRules` with `matchDatasources: ['npm']`?